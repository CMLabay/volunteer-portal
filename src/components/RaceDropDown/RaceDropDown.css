select{
    width: 155px;
    color: #022e59;
    font-size: 15px;
    margin-bottom: 7px;
}
button{
    color: #022e59;
}
