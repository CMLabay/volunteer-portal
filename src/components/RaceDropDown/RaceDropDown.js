import React, { Component } from 'react'
import RacesApiService from '../../services/races-api-service'

export default class RaceDropDown extends Component{
        
    state = { 
        error: null,
        races: [{
          id:'',
          name:'',
        } ]
      }

    componentDidMount(){
        RacesApiService.getRaces()
          .then(res => {
            let races = res;
            this.setState( { races })
          })
          .catch(this.state.error)
      }

    render(){
        const raceList = this.state.races
        return (
            <select name="shiftRaceName"
                    id="shiftRaceName"
                    onChange={this.props.onChange}
                    >
                {raceList.map(race =>
                    <option key={race.id}
                            literal={race.name}
                            value={race.id}>
                            {race.name}
                    </option>
                )}
            </select>
        )
    }
}