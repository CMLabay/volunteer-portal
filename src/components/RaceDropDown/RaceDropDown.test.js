import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import RaceDropDown from './RaceDropDown'
import renderer from 'react-test-renderer';

describe('race drop down component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <RaceDropDown>
                </RaceDropDown>
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the ui as expected', () => {
        const tree = renderer
        .create(
            <BrowserRouter>
            <RaceDropDown></RaceDropDown>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
    })
})