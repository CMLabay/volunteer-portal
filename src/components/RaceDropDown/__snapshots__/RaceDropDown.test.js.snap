// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`race drop down component renders the ui as expected 1`] = `
<select
  id="shiftRaceName"
  name="shiftRaceName"
>
  <option
    literal=""
    value=""
  >
    
  </option>
</select>
`;
